impy/kinematics.py

```diff
@@ -183,6 +305,19 @@ def __init__(
                 self.p1pdg = 2212
             info(20, "Particle 1 is a nucleus.")
 
+        if particle2:
+            particle_input_type = get_particle_input_type(particle2)
+            if particle_input_type == "pdg_id":
+                p2pdg = particle2
+            elif particle_input_type == "tuple":
+                nuc2_prop = particle2
+            elif particle_input_type == "string":
+                p2pdg = FromParticleName.get_pdg(particle2)
+            elif particle_input_type == "composite_target":
+                nuc2_prop = particle2.get_maximum_AZ()
```

Comment:
There is a lot of code duplication here. I would introduce a function `_normalize_particle` (or similar name) which takes the allowed kinds of inputs and returns either a pdgid or a CompositeTarget. Nuclei can be mapped to pdgids. This simplifies the handling downstream and its reduces the code here to just two calls to this function.

Reply:
I agree. It is changed to use `_normalize_particle` to remove duplication.